File: An II/Sem I/MAP/Lab/Lab11/Lab11/Factory/TaskContainerFactory.cs
using Lab11.Enums;
using Lab11.Container;
namespace Lab11.Factory;

public class TaskContainerFactory : Factory
{
    private TaskContainerFactory()
    {
        
    }

    private static readonly TaskContainerFactory instance = new TaskContainerFactory();

    public static TaskContainerFactory getInstance => instance;
    
    public Container.Container createContainer(Strategy containerStrategy)
    {
        return containerStrategy switch
        {
            Strategy.FIFO => new Container.QueueContainer(),
            Strategy.LIFO => new Container.StackContainer(),
            _ => throw new ArgumentOutOfRangeException(nameof(containerStrategy), containerStrategy, null)
        };
    }
}

File: An II/Sem I/MAP/Seminar/Sem1/src/Factory/TaskContainerFactory.java
package Factory;

import Container.Container;
import Container.QueueContainer;
import Container.StackContainer;
import Enums.Strategy;

public class TaskContainerFactory implements Factory {
    //private constructor
    //public static Factory.TaskContainerFactory getInstance()
    private TaskContainerFactory() {

    }

    private static final TaskContainerFactory instance = new TaskContainerFactory();

    public static TaskContainerFactory getInstance(){
        return instance;
    }
    @Override
    public Container createContainer(Strategy strategy) {
        switch (strategy) {
            case Strategy.FIFO -> {
                return new QueueContainer();
            }
            case Strategy.LIFO -> {
                return new StackContainer();
            }
        }
        return null;
    }
}
